src/DDTrace/ScopeManager.php

```diff
@@ -57,6 +76,63 @@ public function deactivate(Scope $scope)
         }
 
         array_splice($this->scopes, $i, 1);
+
+        $span = $scope->getSpan();
+        if ($span instanceof Span && isset($span->ddtrace_scope_activated)) {
+            $span->ddtrace_scope_activated = false;
+        }
+    }
+
+    public function getPrimaryRoot()
+    {
+        $this->getTopScope(); // ensure active
+        return reset($this->scopes) ?: null;
+    }
+
+    public function getTopScope()
+    {
+        $topScope = null; // shut up, phpstan
```

Comment:
Please remove this comment :)

Reply:
well, it's there because the statement itself is redundant... I like to know why seemingly redundant things are there later on?